onnxruntime/core/providers/tensorrt/tensorrt_execution_provider.cc

```diff
@@ -47,9 +47,9 @@ std::string GetEnginePath(const ::std::string& root, const std::string& name) {
 }
 
 std::string GetVecHash(const ::std::vector<int>& vec) {
-  std::size_t ret = 0;
+  std::size_t ret = vec.size();
   for (auto& i : vec) {
-    ret ^= std::hash<uint32_t>()(i);
+    ret ^= i + 0x9e3779b9 + (ret << 6) + (ret >> 2);
```

Comment:
any comments on how this hash function was chosen?

Reply:
The constant number and shifting are used to increase the distance of consecutive hash values and reduce hash conflicts.
Here is a good explanation, https://stackoverflow.com/questions/4948780/magic-number-in-boosthash-combine.